### ISSUE TITLE
clang wrong diagnostic for unused-local-typedef

### ISSUE THREAD

**Issue by lahwaacz** [llvm/llvm-project#59706]
Minimal repro code:
```cpp
template< typename T >
struct A
{};

template< template<typename> class TT >
void
f()
{
   struct inner
   {
      using foo = int;

      foo bar() const
      {
         return {};
      }
   };
}

int main()
{
    f< A >();
}
```

Building with Clang 14 gives a wrong diagnostic:
```console
$ clang++ -Wall -Werror test.cpp
test.cpp:11:13: error: unused type alias 'foo' [-Werror,-Wunused-local-typedef]
      using foo = int;
            ^
1 error generated.
```
Notice that the `foo` type is clearly used as a return type of the `bar` member function. But even if it was not, it should be considered part of the public interface and the diagnostic should not be emitted. Interestingly, removing the template-template parameter `TT` makes the issue go away.

**Comment by xgupta** [comment]
The issue is fixed on the trunk and 15 also - https://godbolt.org/z/WqP1YM7KK. We may close this issue.
